fl4health/utils/partitioners.py

```diff
@@ -70,17 +85,29 @@ def partition_label_indices(self, label: T, label_indices: torch.Tensor) -> Tupl
 
         Returns:
             List[torch.Tensor]: partitioned indices of datapoints with the corresponding label.
+            int: The minimum number of data points assigned to a partition.
+            np.ndarray: The Dirichlet distribution used to partition the data points.
         """
-        total_data_points_with_label = len(label_indices)
-        # These are the percentages of the label indices to be distributed for each partition
-        partition_allocations = np.random.dirichlet(np.repeat(self.beta, self.number_of_partitions))
-        log(
-            INFO,
-            (
-                f"The allocation distribution for label ({str(label)}) is {partition_allocations} "
-                f"using a beta of {self.beta}",
-            ),
-        )
+        if self.prior_distribution is not None:
+            assert (
+                len(self.prior_distribution) == self.n_unique_labels
+            ), "The length of the prior must match the number of labels"
+            partition_allocations = self.prior_distribution[label]
+            log(
+                INFO,
+                f"Using a prior distribution of {self.prior_distribution[label]} to allocate the label ({str(label)})",
+            )
+        if self.beta is not None:
+            total_data_points_with_label = len(label_indices)
```

Comment:
I don't think you want this guy to be here? That is, it's going to cause a problem if you have a prior I think?

Reply:
I am confused which one do you mean exactly?

Reply:
The scope of `total_data_points_with_label` is tied to this if statement but you use it in the code outside of this if to establish `num_samples_per_partition`. If you have a prior defined, then I'm pretty sure `total_data_points_with_label` will be None and when you try to use it later it will throw.